components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -538,64 +541,66 @@ internal sealed class SubscriptionWrapper : ISubscription {
-                            (monitoredItem.Item.Status.Error != null &&
-                            StatusCode.IsNotGood(monitoredItem.Item.Status.Error.StatusCode))) {
-
-                            monitoredItem.Template.MonitoringMode = Publisher.Models.MonitoringMode.Disabled;
-                            noErrorFound = false;
-                            applyChanges = true;
+                    if (_currentlyMonitored != null) {
+                        // do a sanity check
+                        foreach (var monitoredItem in _currentlyMonitored) {
+                            if (monitoredItem.Item.Status.MonitoringMode == Opc.Ua.MonitoringMode.Disabled ||
+                                (monitoredItem.Item.Status.Error != null &&
+                                StatusCode.IsNotGood(monitoredItem.Item.Status.Error.StatusCode))) {
+
+                                monitoredItem.Template.MonitoringMode = Publisher.Models.MonitoringMode.Disabled;
+                                noErrorFound = false;
+                                applyChanges = true;
+                            }
                         }
-                    }
-                    if (applyChanges) {
-                        rawSubscription.ApplyChanges();
-                    }
-                }
-
-                if (activate) {
-                    // Change monitoring mode of all valid items if needed
-                    var validItems = _currentlyMonitored.Where(v => v.Item.Created);
-                    foreach (var change in validItems.GroupBy(i => i.GetMonitoringModeChange())) {
-                        if (change.Key == null) {
-                            continue;
+                        if (applyChanges) {
+                            rawSubscription.ApplyChanges();
                         }
-                        var changeList = change.ToList();
-                        _logger.Information("Set monitoring to {value} for {count} items in subscription "
-                            + "'{subscription}'/'{sessionId}'.",
-                            change.Key.Value,
-                            change.Count(),
-                            Id,
-                            Connection.CreateConnectionId());
-
-                        var itemsToChange = changeList.Select(t => t.Item).ToList();
-                        var results = rawSubscription.SetMonitoringMode(change.Key.Value, itemsToChange);
-                        if (results != null) {
-                            var erroneousResultsCount = results
-                                .Count(r => (r == null) ? false : StatusCode.IsNotGood(r.StatusCode));
 
-                            // Check the number of erroneous results and log.
-                            if (erroneousResultsCount > 0) {
-                                _logger.Warning("Failed to set monitoring for {count} items in subscription "
+                        if (activate) {
```

Comment:
Please move the logic under this `if` out of `else` case on line `543`.

Reply:
fixed